src/librustc/mir/mod.rs

```diff
@@ -2873,10 +2877,11 @@ pub struct GeneratorLayout<'tcx> {
     pub storage_conflicts: BitMatrix<GeneratorSavedLocal, GeneratorSavedLocal>,
 
     /// Names and scopes of all the stored generator locals.
-    /// NOTE(tmandry) This is *strictly* a temporary hack for codegen
-    /// debuginfo generation, and will be removed at some point.
-    /// Do **NOT** use it for anything else, local information should not be
-    /// in the MIR, please rely on local crate HIR or other side-channels.
+    //
+    // NOTE(tmandry) This is *strictly* a temporary hack for codegen
```

Comment:
Btw, I thought about this on a second pass. I think it is intentional for this to be a doc comment so it is noticed when reading the internal rustdocs if one searches for it there.

Reply:
`NOTE`/`FIXME`/`HACK` should never be a doc comment though. It's fine as is.

Reply:
Okay, I compromised... see the commit. :-)